AddResourceInfo.cmake

```diff
@@ -0,0 +1,78 @@
+#CMAKE_CURRENT_LIST_DIR is the directory containing this file when
+#referenced here, at top-level scope, when this file is included.
+#From within a macro, CMAKE_CURRENT_LIST_DIR is the directory that
+#contains the file being processed and invoking the function.  So,
+#to use this files directory within a macro, we need to save it at
+#top level scope at include time.
+set(THIS_FILE_DIR ${CMAKE_CURRENT_LIST_DIR})
+
+#This macro "finishes" the file attributes that should've been already set by the CMakeLists.txt that calls
+#add_resource_info. It does this by appending the Git commit hash to the original name (so Windows knows when
+#a different version is present), creates the original name in the first place (which should always just be
+#the filename), sets the year release to the current year, and appends the architecture to the description.
+macro(set_common_attributes_windows PROJECT fileExtension)
+get_git_head_revision(GIT_REFSPEC GIT_COMMIT_HASH)
+string(TIMESTAMP CURRENT_YEAR "%Y")
+if(CMAKE_SIZEOF_VOID_P EQUAL 4)
+    set(ARCHITECTURE_DESCRIPTION "32-bit")
+else()
+    set(ARCHITECTURE_DESCRIPTION "64-bit")
+endif()
+set( FILE_YEAR_RELEASE ${CURRENT_YEAR})
+set( FILE_DESCRIPTION "${FILE_DESCRIPTION} (${ARCHITECTURE_DESCRIPTION})")
+set( FILE_ORIGINAL_NAME "${PROJECT}.${fileExtension} ${GIT_COMMIT_HASH}" )
+endmacro()
+
+#This macro (currently Windows-only) creates a .RC file that holds all of the application/library's file details. This RC
+#file is then appended to the source list for the target.
+#
+#Here are the following variables the macro expects are properly set before executing:
+#FILE_VERSION_MAJOR        - The major version of the file
+#FILE_VERSION_MINOR        - The minor version of the file
+#FILE_VERSION_PATCH        - The patch version of the file
+#FILE_DESCRIPTION          - A description of the file
+#FILE_PRODUCT_NAME         - A name of the product. Usually just the fully-expanded name for the application/library.
```

Comment:
I think it would be far better for this macro to take a long list of parameters and set these itself instead of requiring the user to set them, for several reasons:

- It's very non-obvious in the calling code that these are used in this macro (it looks like a bunch of variables are set and never used)
- This is exposing the implementation details of the macro, and the .rc file, all over our repositories
- If we wanted to change how something is represented in this resource info, it would be better to have the ability to do it in one place, rather than need to change it again in every repository.

Reply:
Yeah I like this idea!